Photomosaic task

mradzikowski opened the pull request:
Added photomosaic task's algorithm, the order of taking photos is important (TOP -> LEFT -> FRONT -> RIGHT -> BACK). Put constants to the constants folder with vision.

mradzikowski committed:
Add algorithm for photomosaic task

mradzikowski committed:
Add constants for vision regarding to photomosaic task

mradzikowski committed:
Update photomosaic_photo.py

mradzikowski committed:
Update photomosaic_photo.py

mradzikowski committed:
Update photomosaic_photo.py

mradzikowski committed:
Update photomosaic_photo.py

mradzikowski committed:
Update photomosaic_photo.py

mradzikowski committed:
Update photomosaic_photo.py

mradzikowski committed:
Change ndarrays to np.ndarrays

mradzikowski committed:
Change ndarrays to np.ndarrays

mradzikowski committed:
Update mussels.py

mradzikowski committed:
Merge branch 'photomosaic-task' of https://github.com/ncl-ROVers/surface into photomosaic-task

mradzikowski committed:
Change typo

mradzikowski committed:
Specify version of opencv-python

TheCodeSummoner reviewed (changes requested):
Generally:

- No underscore imports, please
- More comments!
- Better docs!
- Add a README with an example!
- Try to new-line after code with comments (my preference really, but I think it looks better):
- I will review again after these fixes :)

```
# some comment
some_code()

# another comment
more_code()
```

For commits, consider rebasing. In PyCharm:

![image](https://user-images.githubusercontent.com/36332694/125169115-08e6d400-e1a9-11eb-992a-7aefcbf92be4.png)

You can fixup, drop, reword commits (fixup means merge to the above one):
![image](https://user-images.githubusercontent.com/36332694/125169141-1b610d80-e1a9-11eb-8693-4608009513ab.png)

Keep in mind you will need to force push (so careful!!! We can do it together if it's easier / you feel safer, at the end, once the branch is ready to be merged):

![image](https://user-images.githubusercontent.com/36332694/125169174-3df32680-e1a9-11eb-87f1-90c2bcf3b377.png)

mradzikowski committed:
Change imports not to be underscored

MOSAIC_HEIGHT is referred to the result picture

mradzikowski committed:
Change opencv installation version to be default

Change in mussels.py parsing to integers by function int() instead of astype()

mradzikowski committed:
Refactor code to be more clear and consistent

Added logger and got rid of print statements. Changed imports to not be underscored. Added documentation about _cut_images function, _find_rectangle, _get_contours, _combine_images, _color_detect. _get_key function is to help with combining images and getting colors of interest to match while combining.

mradzikowski committed:
Add type-hinting and change parsing to integers

Changed parsing to be compatible with newer versions of OpenCV
